src/site/content/en/blog/asyncify/index.md

```diff
@@ -0,0 +1,516 @@
+API.
+
+When compiling such code, you need to tell Emscripten to activate the Asyncify feature. Do that by
+passing `-s ASYNCIFY` as well as [`-s ASYNCIFY_IMPORTS=[func1,
+func2]`](https://emscripten.org/docs/porting/asyncify.html#more-on-asyncify-imports) with a
+comma-separated list of functions that might be asynchronous.
+
+```shell
+emcc -O2 \
+    -s ASYNCIFY \
+    -s ASYNCIFY_IMPORTS=[async_sleep] \
+    ...
+```
+
+This lets Emscripten know that any calls to those functions might require saving and restoring the
+state, so the compiler will inject a supporting code around such calls.
+
+Now, when you execute this
+code in the browser you'll see a seamless output log like you'd expect, with B coming after a short
+delay after A.
+
+```text
+A
+B
+```
+
+You can [return values from
+Asyncify](https://emscripten.org/docs/porting/asyncify.html#returning-values) functions too. What
+you need to do is return the result of `handleSleep`, and pass the result to the `wakeUp()`
+callback. For example, if, instead of reading from a file, you want to fetch some numeric answer
+from a remote resource, you can use a snippet like below to issue a request, suspend the C code, and
+resume once the response body is retrieved—all done seamlessly as if the call was synchronous.
+
+```js
+EM_JS(int, get_answer, (), {
+     return Asyncify.handleSleep(wakeUp => {
+        fetch("answer.txt")
+            .then(response => response.text())
+            .then(text => wakeUp(Number(text)));
+    });
+});
+puts("Getting answer...");
+int answer = get_answer();
+printf("Answer is %d\n", answer);
+```
+
+In fact, for Promise-based APIs like `fetch()`, you can even combine Asyncify with JavaScript
+async-await feature instead of using the callback-based API. For that, instead of
+`Asyncify.handleSleep()`, call `Asyncify.handleAsync()`. Then, instead of having to schedule a
+`wakeUp()` callback, you can pass an `async` JavaScript function and use `await` and `return`
+inside, making code look even more natural and synchronous, while not losing any of the benefits of
+the asynchronous I/O.
+
+```js
+EM_JS(int, get_answer, (), {
+     return Asyncify.handleAsync(async () => {
+        let response = await fetch("answer.txt");
+        let text = await response.text();
+        return Number(text);
+    });
+});
+
+int answer = get_answer();
+```
+
+#### Awaiting complex values
+
+But this example still limits you only to numbers. What if you want to implement our original
+example, where we tried to get a user's name from a file as a string? Well, you can do that too!
+
+Emscripten provides a feature called
+[Embind](https://emscripten.org/docs/porting/connecting_cpp_and_javascript/embind.html), that allows
+you to handle conversions between JavaScript and C++ values. It has support for Asyncify as well, so
+you can call `.await()` on external `Promise`s and it will act just like `await` in an async-await
```

Comment:
```suggestion
you can call `await()` on external `Promise`s and they will act just like `await` in async-await
```

Reply:
1) Isn't `.` better because it shows that it's a method not a bare function like in all other examples we've shown so far?
2) it's "it", as in `.await()`, not "they" as in "Promises".

Reply:
1. This is the convention. I've never heard of it causing confusion.
2. My bad. Fixed.